Source/OnboardingContentViewController.m

```diff
@@ -29,10 +30,17 @@
 
 @interface OnboardingContentViewController ()
 
+@property (nonatomic, strong) MPMoviePlayerController *moviePlayerController;
+@property (nonatomic, strong) NSURL *videoURL;
+
 @end
 
 @implementation OnboardingContentViewController
 
+- (void)dealloc {
+    [self removeObserver:self forKeyPath:UIApplicationWillEnterForegroundNotification];
```

Comment:
Very cool - so do you need to call that on `self, as you did,` or `[NSNotificationCenter defaultCenter] `?